src/mlpack/core/data/split_data.hpp

```diff
@@ -368,6 +378,236 @@ Split(const arma::Mat<T>& input,
                          std::move(testData));
 }
 
+/**
+ * Given an input dataset and labels, split into a training set and test set.
+ * Example usage below.  This overload places the split dataset into the four
+ * output parameters given (trainData, testData, trainLabel, and testLabel).
+ *
+ * The input dataset must be of type arma::field and have a single row.
```

Comment:
Right, so just so I'm sure I understand, you mean that the input dataset should be an `arma::field` with one row and number of columns equal to the number of datasets in the point?  Or one row, one column, and number of slices equal to the number of points?  I'm not entirely sure what the meaning is, or what I should expect the shape of the output to be.

Reply:
Hmm.. This is a bit vague. I will fix it too. 
Most probably I will finish this one by today. Just a couple of test cases are left. This time I will try to avoid such silly errors :)